### PR TITLE
MINOR: Refactor `MetricsIntegrationTest`

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cadonna** [comment]
Call for review @abbccdda @ableegoldman @vvcephei @mjsax @bbejeck @guozhangwang

**Comment by bbejeck** [comment]
@cadonna failures are related - checkstyle

**Review by bbejeck — COMMENTED** [review]
Changes overall look good to me, just need to resolve checkstyle issues.

**Comment by cadonna** [comment]
Fixed checkstyle issues.

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java [hunk 1]
```diff
@@ -244,241 +232,248 @@ public void testStreamMetric() throws Exception {
...
+        checkMetricByName(listMetricStore, RANGE_TOTAL, 2);
+        checkMetricByName(listMetricStore, FLUSH_RATE, 2);
+        checkMetricByName(listMetricStore, FLUSH_TOTAL, 2);
+        checkMetricByName(listMetricStore, RESTORE_RATE, 2);
+        checkMetricByName(listMetricStore, RESTORE_TOTAL, 2);
+    }
+
+    private void checkMetricsDeregistration() {
+        final List<Metric> listMetricAfterClosingApp = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream()
+            .filter(m -> m.metricName().group().contains(STREAM_STRING)).collect(Collectors.toList());
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java]: as above

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java [hunk 2]
```diff
@@ -244,241 +232,248 @@ public void testStreamMetric() throws Exception {
...
-        testMetricByName(listMetricTask, COMMIT_RATE, 5);
-        testMetricByName(listMetricTask, COMMIT_TOTAL, 5);
-        testMetricByName(listMetricTask, RECORD_LATENESS_AVG, 4);
-        testMetricByName(listMetricTask, RECORD_LATENESS_MAX, 4);
+        checkMetricsDeregistration();
     }
 
     private void checkThreadLevelMetrics() {
         final List<Metric> listMetricThread = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream()
             .filter(m -> m.metricName().group().equals(STREAM_THREAD_NODE_METRICS)).collect(Collectors.toList());
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java]: nit: move .collect to new line

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java [hunk 3]
```diff
@@ -244,241 +232,248 @@ public void testStreamMetric() throws Exception {
...
+        checkMetricByName(listMetricThread, TASK_CREATED_TOTAL, 1);
+        checkMetricByName(listMetricThread, TASK_CLOSED_RATE, 1);
+        checkMetricByName(listMetricThread, TASK_CLOSED_TOTAL, 1);
+        checkMetricByName(listMetricThread, SKIPPED_RECORDS_RATE, 1);
+        checkMetricByName(listMetricThread, SKIPPED_RECORDS_TOTAL, 1);
+    }
+
+    private void checkTaskLevelMetrics() {
+        final List<Metric> listMetricTask = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream()
+            .filter(m -> m.metricName().group().equals(STREAM_TASK_NODE_METRICS)).collect(Collectors.toList());
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java]: as above

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java [hunk 4]
```diff
@@ -244,241 +232,248 @@ public void testStreamMetric() throws Exception {
...
+        checkMetricByName(listMetricTask, COMMIT_LATENCY_MAX, 5);
+        checkMetricByName(listMetricTask, COMMIT_RATE, 5);
+        checkMetricByName(listMetricTask, COMMIT_TOTAL, 5);
+        checkMetricByName(listMetricTask, RECORD_LATENESS_AVG, 4);
+        checkMetricByName(listMetricTask, RECORD_LATENESS_MAX, 4);
     }
 
     private void checkProcessorLevelMetrics() {
         final List<Metric> listMetricProcessor = new ArrayList<Metric>(kafkaStreams.metrics().values()).stream()
             .filter(m -> m.metricName().group().equals(STREAM_PROCESSOR_NODE_METRICS)).collect(Collectors.toList());
```
Comment (ableegoldman) [inline streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java]: as above

**Comment by ableegoldman** [comment]
Thanks @cadonna! Few small nits but otherwise LGTM

File: streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java (hunk 2 above)
Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/integration/MetricsIntegrationTest.java]: Ack

**Review by bbejeck — APPROVED** [review]
LGMT modulo comments from @ableegoldman

**Comment by bbejeck** [comment]
Merged #6930 into trunk

**Pull request merged**